### PR TITLE
Make xhr request in getImage

### LINKED ISSUE #1470: setStyle does not re-fetch sprite image
When a sprite image changes and `setStyle` is called, a new image should be fetched in https://github.com/mapbox/mapbox-gl-js/blob/master/js/util/browser/ajax.js#L51-L69.

### PR DESCRIPTION
**Pull request by edenh**
close #1470

:eyes: @jfirebaugh @bhousel

### CODE REVIEW

**Comment by jfirebaugh** [comment]
Cool approach! Does this work in all the supported browsers?

File: js/util/browser/ajax.js [hunk 1]
```diff
@@ -49,23 +49,38 @@ function sameOrigin(url) {
 }
 
 exports.getImage = function(url, callback) {
-    var img = new Image();
-    if (!sameOrigin(url)) {
-        img.crossOrigin = 'Anonymous';
-    }
-    img.onload = function() {
-        callback(null, img);
+    var xhr = new XMLHttpRequest();
```
Comment (jfirebaugh) [inline js/util/browser/ajax.js]: Use `exports.getArrayBuffer` internally here to avoid some code duplication.

**Comment by edenh** [comment]
Sprites are note being rendered in IE 11 with no errors despite good support for [`createObjectURL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL#Browser_compatibility)

**Comment by edenh** [comment]
Ref https://github.com/mapbox/mapbox-gl-js/issues/1475

**Comment by gertcuykens** [comment]
When I change `map.setStyle('mapbox://styles/...`)` in v0.12.2 from satellite raster to street vector

I get in chrome `GET blob:http%3A//localhost%3A8080/8a9eafc8-68aa-4334-8801-5df28ed14022 404 (Not Found)` and the new style doesn't show tiles.

The url is ok when I use it in `new mapboxgl.Map({style:`mapbox://styles/...`})`

PS satellite raster never shows custom maki icons.

**Pull request merged**